src/content_mapblock.cpp

```diff
@@ -30,22 +30,43 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "log.h"
 #include "noise.h"
 
+// Distance of light extrapolation (for oversized nodes)
+// After this distance, it gives up and considers light level constant
+#define SMOOTH_LIGHTING_OVERSIZE 1.0
+
+struct LightFrame
+{
+	f32 lightsA[8];
+	f32 lightsB[8];
+	u8 light_source;
+};
+
+static const v3s16 light_dirs[8] = {
+	v3s16(-1, -1, -1),
+	v3s16(-1, -1,  1),
+	v3s16(-1,  1, -1),
+	v3s16(-1,  1,  1),
+	v3s16( 1, -1, -1),
+	v3s16( 1, -1,  1),
+	v3s16( 1,  1, -1),
+	v3s16( 1,  1,  1),
+};
 
 // Create a cuboid.
 //  collector - the MeshCollector for the resulting polygons
 //  box       - the position and size of the box
 //  tiles     - the tiles (materials) to use (for all 6 faces)
 //  tilecount - number of entries in tiles, 1<=tilecount<=6
-//  c         - vertex colour - used for all
+//  c         - vertex colours. The order is the same as in light_dirs
 //  txc       - texture coordinates - this is a list of texture coordinates
 //              for the opposite corners of each face - therefore, there
 //              should be (2+2)*6=24 values in the list. Alternatively, pass
 //              NULL to use the entire texture for each face. The order of
 //              the faces in the list is up-down-right-left-back-front
 //              (compatible with ContentFeatures). If you specified 0,0,1,1
 //              for each face, that would be the same as passing NULL.
-void makeCuboid(MeshCollector *collector, const aabb3f &box,
-	TileSpec *tiles, int tilecount, video::SColor &c, const f32* txc)
+static void makeSmoothLightedCuboid(MeshCollector *collector, const aabb3f &box,
+	TileSpec *tiles, int tilecount, const video::SColor c[8], const f32* txc)
```

Comment:
what @rubenwardy said is not actually true, adding the array size to the call **does not** protect from wrongly-sized array.
This code:
```c++
#include <iostream>

void thing(const int array[8]) {
	std::cout << "first element: " << array[0] << std::endl;
}

int main() {
	int a1[8] = {0,0,0,0,0,0,0,0};
	thing(a1);

	int a2[2] = {1,1};
	thing(a2); //too small

	int a3[16] = {2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2};
	thing(a3); //too big

	int a4 = 3;
	thing(&a4); // not an array
}
```
...compiles without **any** warnings using `-Wall -Wextra`

tl;dr change it back to `const video::SColor *c` to avoid confusion